packages/lit-dev-content/site/articles/article/lit-cheat-sheet.md

```diff
@@ -0,0 +1,1118 @@
+
+This is a new feature recently added to some browsers.
+
+Please check [browser compatibility on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with#browser_compatibility).
+
+Due to the newness of this feature, the following example uses JavaScript and
+may not work in all in certain browsers.
+
+{% endaside %}
+
+{% playground-ide "articles/lit-cheat-sheet/import-attributes", true %}
+
+{% aside "positive" %}
+
+Some better-supported alternatives to this approach may include:
+
+- A build tool plugin that transforms your CSS imports into Lit `CSSResult`
+  similar to
+  [`rollup-plugin-lit-css`](https://www.npmjs.com/package/rollup-plugin-lit-css)
+- Using a bundler tool to transform your CSS imports into strings and then using
+  Constructable Stylesheets
+- Using a `<link rel="stylesheet" href="...">` in your template, but this will
+  cause FOUC.
+
+{% endaside %}
+
+**Related Documentation & Topics:**
+
+- [Import Attributes](https://v8.dev/features/import-attributes) (external)
+- [Import Attributes Browser Compatiblity Table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with#browser_compatibility) (external)
+- [Proposal Import Attributes](https://github.com/tc39/proposal-import-attributes) (external)
+- [Import JSON, CSS and more with import attributes](https://fullystacked.net/import-attributes/) (external)
+- [Using CSS Module Scripts to import stylesheets](https://web.dev/articles/constructable-stylesheets) (external)
+  - **NOTE:** The syntax of the import assertions proposal in this article was
+    deprecated in favor of import attribtues which uses `with` rather than
+    `assert`.
+- Build Tools
+  - [`rollup-plugin-lit-css`](https://www.npmjs.com/package/rollup-plugin-lit-css)
+  - [`esbuild-plugin-lit-css`](https://www.npmjs.com/package/esbuild-plugin-lit-css)
+  - [`lit-css-loader`](https://www.npmjs.com/package/lit-css-loader)
+  - [`vite-plugin-lit-css`](https://www.npmjs.com/package/vite-plugin-lit-css)
+
+## Shadow DOM
+
+### What does Shadow DOM do?
+
+- Scopes styles to the shadow root
+- Scopes the DOM to the shadow root
+  - Can't be targeted by querySelector calls from outside the shadow root
+- Enables slotting content with the `<slot>` element
+- Exposes an API for CSS with CSS Custom Properties and CSS Shadow Parts
+
+**Related Documentation & Topics:**
+
+- [DOM Encapsulation](/docs/components/rendering/#dom-encapsulation)
+- [Working with Shadow DOM](/docs/components/shadow-dom/)
+- [How to style your Lit elements](https://www.youtube.com/watch?v=Xt7blcyuw5s) (video)
+
+### Turn off Shadow DOM
+
+You can turn off the Shadow DOM by overriding the `createRenderRoot()` method and setting the render root to the element itself.
+
+{% playground-ide "articles/lit-cheat-sheet/turn-off-shadow-dom", true %}
+
+{% aside "warn" %}
+
+By turning off shadow DOM you lose the benefits of encapsulation, DOM scoping, and `<slot>` elements.
```

Comment:
```suggestion
This is generally not recommended, but it may sometimes be worthwhile for integration with older systems or libraries that may not be updated to work with Shadow DOM. 

By turning off shadow DOM you lose the benefits of encapsulation, DOM scoping, and `<slot>` elements.
```